### ISSUE TITLE
Sync-UDElement can cause Hide-UDModal to fail (Error 404)

### ISSUE THREAD

**Issue by DataTraveler1** [ironmansoftware/powershell-universal#1205]
### Steps to Reproduce

This issue was reported internally. Code to reproduce is below. An animated gif is further below. The error demonstrated here (when viewed through Dev Tools) reveals that _Error 404_ are happening when Hide-UDModal fails.

❗ The modal will close if the Sync-UDElement command from the below example code is removed

> if the modal is nested in the scriptblock of a UDDynamic and the UDDynamic is refreshed while the modal is up, Hide-UDModal doesn’t work anymore.

```
New-UDDashboard -Title 'Issue #1204 Demonstration' -Content {
    New-UDDynamic -Id 'table' -Content {
        New-UDButton -Text "Start demo" -OnClick {
            Show-UDModal -Content {
                New-UDButton -text 'Begin work' -OnClick {
                    Hide-UDModal
                    Show-UDModal -Persistent -Content {
                        New-UDAlert -Text 'Getting work done...'
                        New-UDProgress
                    }
                    Start-Sleep -Seconds 3                    
                    Sync-UDElement -Id 'table' #<--- remove this to workaround the bug
                    Hide-UDModal
                    Show-UDModal -Persistent -Content {
                        New-UDButton -Text 'You can''t close this' -OnClick {
                            Hide-UDModal
                        }
                    }
                }
            }
        }
    }
}
```

### Expected behavior

```console
The modal should close when the close button is clicked
```

### Actual behavior

```console
The modal remains after the close button is clicked (apparently due to Sync-UDElement)
```

### Environment data

PSU 2.11.2 (nightly build)

### Visuals

![Issue1204](https://user-images.githubusercontent.com/537287/170842961-82e92ccb-590d-42cc-82ae-9e06e9b98eda.gif)

**Comment by DataTraveler1** [comment]
##### Below figure shows when the modal is working correctly (by commenting out the line with Sync-UDElement)![image](https://user-images.githubusercontent.com/537287/170852684-b27d993c-c5f3-4515-8396-11849817f341.png)

##### Below figure shows when the modal is not working correctly (it requests a URL that doesn't exist)
![image](https://user-images.githubusercontent.com/537287/170852701-c6d1a7ea-0b23-4c71-b786-d5757b060e11.png)

**Comment by DataTraveler1** [comment]
@adamdriscoll - Gentle (internal) query if it would be possible for this milestone to also be added to the 2.x branch

**Comment by adamdriscoll** [comment]
I've had to roll this change back because it was causing other issues. I'll reevaluate after our 3.0.0 and 2.12.0 release.

**Comment by Entech-Ramon** [comment]
Any ballpark estimate on when this might be added or is there some kind of work around I can do? I can't update to the latest version of 3 or 2 and still on 1.5.21, really just waiting for this bug to be worked out. Thanks.

**Comment by Entech-Ramon** [comment]
@adamdriscoll, any idea on when this fix can be implemented?

**Comment by adamdriscoll** [comment]
@Entech-Ramon - I'll make sure this is fixed for our 3.1 release next week
